src/Api/Response/ResponseParser.php

```diff
@@ -198,4 +198,22 @@ protected function parseProducts(\SimpleXMLElement $data)
 
         return $products;
     }
-}
\ No newline at end of file
+
+    /**
+     * @param \SimpleXMLElement $data
+     * @return array
+     */
+    protected function getFilters($data)
+    {
+        $mainFilters = $this->filtersParser->parse($data->filters->main);
+        $otherFilters = $this->filtersParser->parse($data->filters->other);
+
+        if (!empty($mainFilters)) {
+            foreach ($mainFilters as $mainFilterKey => $mainFilterValue) {
```

Comment:
I don't like the idea that we have to iterate over all main filters again. How about rewriting `\Findologic\Api\Response\Parser\FiltersParser::parse` so that we can add the `isMain` flag there?
In `ResponseParser::parse` you could then do `$response->setData(Response::DATA_FILTERS, $this->filtersParser->parse($data->filters));`.